internal/services/datafactory/data_factory_dataset_snowflake_resource.go

```diff
@@ -121,27 +121,49 @@ func resourceDataFactoryDatasetSnowflake() *pluginsdk.Resource {
 							Type:     pluginsdk.TypeString,
 							Optional: true,
 							ValidateFunc: validation.StringInSlice([]string{
-								"Byte",
-								"Byte[]",
-								"Boolean",
-								"Date",
-								"DateTime",
-								"DateTimeOffset",
-								"Decimal",
-								"Double",
-								"Guid",
-								"Int16",
-								"Int32",
-								"Int64",
-								"Single",
-								"String",
-								"TimeSpan",
```

Comment:
we cannot remove these until 3.0 - can we add a comment to that affect and leave these in?

Reply:
Thanks for reviewing my PR @katbyte. Why we can't replace those with the right data types of Snowflake? Don't think it works as it is now...

Reply:
someone might have a confid with it somewhere, that maybe hasn't run for a while and used to work and the API is returning these old values - we don't want tf to suddenly fail so we need to keep them around until 3.0 (which is in the next 6 months)